pkg/apis/studycontroller/v1alpha1/studycontroller_types.go

```diff
@@ -0,0 +1,172 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//runtime "k8s.io/apimachinery/pkg/runtime"
+
+	pb "github.com/kubeflow/katib/pkg/api"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// StudyControllerSpec defines the desired state of StudyController
+type StudyControllerSpec struct {
+	StudySpec         *StudySpec         `json:"studySpec,omitempty"`
+	WorkerSpec        *WorkerSpec        `json:"workerSpec,omitempty"`
+	SuggestionSpec    *SuggestionSpec    `json:"suggestionSpec,omitempty"`
+	EarlyStoppingSpec *EarlyStoppingSpec `json:"earlyStoppingSpec,omitempty"`
+}
+
+// StudyControllerStatus defines the observed state of StudyController
+type StudyControllerStatus struct {
+	// Represents time when the TFJob was acknowledged by the TFJob controller.
+	// It is not guaranteed to be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+
+	// Represents time when the TFJob was completed. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+
+	// Represents last time when the TFJob was reconciled. It is not guaranteed to
+	// be set in happens-before order across separate operations.
+	// It is represented in RFC3339 form and is in UTC.
+	LastReconcileTime *metav1.Time `json:"lastReconcileTime,omitempty"`
+
+	State   State      `json:"state,omitempty"`
```

Comment:
In k8s community I think we prefer conditions instead of phase. PTAL https://github.com/kubernetes/kubernetes/issues/7856

Reply:
Thanks, I will fix it.